### PR TITLE
refactor(api-client, react-api-client, app): separate analyses from protocol resource

### LINKED ISSUE #10026: app: split protocol analyses into Analysis and AnalysisSummary object
## Overview

For performance reasons, the robot-server is changing how analyses are returned in HTTP requests from the `/protocols` and `/protocols/:id` endpoints. See issue #9965 and PR #9992 for more details.

## Acceptance criteria

- [ ] The app uses `GET /protocols` and `GET /protocols/:id` to answer the question "is the robot's protocol analysis completed?"
- [ ] The app uses `GET /protocols/:id/analyses` to answer the question "what was the robot's full protocol analysis?"

### Before

```js
// GET /procotols
{
  data: [
    {
      id: 'protocol-id-abc',
      analyses: [
        {
          id: 'analysis-id-xyz',
          status: 'completed',
          result: 'ok',
          commands: [ ... ],  // big commands list
        }
      ],
      // ...
    }
  ]
}
```

### After

```js
// GET /procotols
{
  data: [
    {
      id: 'protocol-id-abc',
      analysisSummaries: [
        {
          id: 'analysis-id-xyz',
          status: 'completed',
        }
      ],
      // ...
    }
  ]
}

// GET /procotols/:id/analyses (new endpoint)
{
  data: [
    {
      id: 'analysis-id-xyz',
      status: 'completed',
      result: 'ok',
      commands: [ ... ],  // big commands list
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

The protocol resource returned from the robot server's http api includes analysis summaries in place
of full weight analysis record for performance reasons. This updates the client-side structures
that relied on the old resource. In order to support the new structure, this adds a new function and
hook to the api-client and react-api-client respectively, for fetching the full analyses associated
with a protocol record.  In order to make these changes, much of the legacy 'run' and 'upload'
pages of the app would have had to be updated. Because these views are slated for immediate removal
ahead of 5.1.0's release, they have been stubbed and removal has begun where functionality in the
new app is sufficient on its own.

Closes #10026 

# Changelog

- add `getProtocolAnalyses` and `useProtocolAnalysesQuery`
- consolidate `useProtocolForRun` into existing `useProtocolDetailsForRun`
- remove legacy app logic that references old protocol record `analyses` key
- deprecate and slate for removal leftover references to `analyses` in legacy app 

# Review requests

- confirm all protocol run-related views in function as expected in the app

# Risk assessment
high; this starts the removal process of legacy app code

### CODE REVIEW

File: app/src/organisms/Devices/hooks/useProtocolDetailsForRun.ts [hunk 1]
```diff
@@ -11,19 +17,40 @@ export interface ProtocolDetails {
...
-  const protocolRecord = useProtocolForRun(runId)
-
-  const protocolAnalysis = protocolRecord?.data.analyses
-  if (protocolAnalysis != null) {
-    const lastProtocolAnalysis = protocolAnalysis[protocolAnalysis.length - 1]
-    if (lastProtocolAnalysis.status === 'completed') {
-      protocolData = schemaV6Adapter(lastProtocolAnalysis)
+  const { data: runRecord } = useRunQuery(runId)
+  const protocolId = runRecord?.data?.protocolId ?? null
+  const enableProtocolPolling = React.useRef<boolean>(true)
```
Comment (mcous) [inline app/src/organisms/Devices/hooks/useProtocolDetailsForRun.ts]: Should this be a `useState` to ensure the `useEffect` below triggers a re-render to actually disable the poll? Or are we comfortable with the poll stopping whenever the component happens to re-render?

**Review by mcous — APPROVED** [review]
Client <> server interaction code looks good from CPX's standpoint. Can't speak to the greater app-reorg stuff

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/hooks/__tests__/useIntroInfo.test.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { createStore, Store } from 'redux'
 import { renderHook } from '@testing-library/react-hooks'
 import { getPipetteNameSpecs, PipetteName } from '@opentrons/shared-data'
 import { getPipetteMount } from '../../../utils/getPipetteMount'
-import { useProtocolDetails } from '../../../../RunDetails/hooks'
+import { useProtocolDetailsForRun } from '../../../../Devices/hooks/useProtocolDetailsForRun'
 import * as hooks from '..'
```
Comment (brenthagen) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/hooks/__tests__/useIntroInfo.test.tsx]: i had the same test failure in my LPC update branch, this should fix:
```suggestion
import { useIntroInfo } from '../useIntroInfo'
```

**Review by brenthagen — APPROVED** [review]
LGTM !

**Pull request merged**